mycroft/util/__init__.py

```diff
@@ -28,6 +28,52 @@
 __author__ = 'jdorleans'
 
 
+def resolve_resource_file(res_name):
+    """Convert a resource into an absolute filename.
+
+    Resource names are in the form: 'filename.ext'
+    or 'path/filename.ext'
+
+    The system wil look for ~/.mycroft/res_name first, and
+    if not found will look at /opt/mycroft/res_name,
+    then finally it will look for res_name in the 'mycroft/res'
+    folder of the source code package.
+
+    Example:
+    With mycroft running as the user 'bob', if you called
+        resolve_resource_file('snd/beep.wav')
+    it would return either '/home/bob/.mycroft/snd/beep.wav' or
+    '/opt/mycroft/snd/beep.wav' or '.../mycroft/res/snd/beep.wav',
+    where the '...' is replaced by the path where the package has
+    been installed.
+
+    Args:
+        res_name (str): a resource path/name
+    """
+
+    # First look for fully qualified file (e.g. a user setting)
+    if os.path.isfile(res_name):
+        return res_name
+    
+    # Now look for ~/.mycroft/res_name (in user folder)
+    filename = os.path.expanduser("~/.mycroft/"+res_name)
+    if os.path.isfile(filename):
+        return filename
+
+    # Next look for /opt/mycroft/res/res_name
+    filename = os.path.expanduser("/opt/mycroft/"+res_name)
+    if os.path.isfile(filename):
+        return filename
+
+    # Finally look for it in the source package
+    filename = dirname(os.path.abspath(__file__))+"../res/"+res_name
```

Comment:
this code isn't upping the directory level and because that the isn't founding the file
a solution could be `os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'res', res_name))`